### PR TITLE
Enable Lint Rule: unused-parameter

### PR DESCRIPTION
**Pull request by FlamingSaint**
## Which problem is this PR solving?
-  Partial Fix for #5506

## Description of the changes
- Enabled unused-parameter in revive linter
- Removed or replaced unused parameters with ( _ )
- Added comments for unclear parameters 

## How was this change tested?
- `make lint` `make test`

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`
  -

### CODE REVIEW

File: cmd/agent/app/processors/thrift_processor_test.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ type failingHandler struct {
 	err error
 }
 
-func (h failingHandler) Process(_ context.Context, iprot, oprot thrift.TProtocol) (success bool, err thrift.TException) {
+func (h failingHandler) Process(context.Context, thrift.TProtocol, thrift.TProtocol) (success bool, err thrift.TException) {
```
Comment (yurishkuro) [inline cmd/agent/app/processors/thrift_processor_test.go:124]: I don't think this is a great approach. It's ok in some cases to replace name with _ or even drop it when the type is reasonably unique, like Context or zap.Logger. But when it's either very common types like `string` or repeated type as in this case, a better approach is change the name into a comment, e.g.
```suggestion
func (h failingHandler) Process(context.Context, /* in */ thrift.TProtocol, /* out */ thrift.TProtocol) (success bool, err thrift.TException) {
```

Comment (FlamingSaint) [reply cmd/agent/app/processors/thrift_processor_test.go:124]: Understood. I will make the changes.

File: cmd/collector/app/sanitizer/service_name_sanitizer_test.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ func (d *fixedMappingCache) Get(key string) string {
 	return k
 }
 
-func (*fixedMappingCache) Put(key string, value string) error {
+func (*fixedMappingCache) Put(string /* key */, string /* value */) error {
```
Comment (yurishkuro) [inline cmd/collector/app/sanitizer/service_name_sanitizer_test.go:43]: why did you change the order? Seems it would've been better to just comment the name in place, to keep the overall Go syntax of `name type`

Comment (FlamingSaint) [reply cmd/collector/app/sanitizer/service_name_sanitizer_test.go:43]: I thought i would maintain consistency everywhere by using it after the type.

Comment (FlamingSaint) [reply cmd/collector/app/sanitizer/service_name_sanitizer_test.go:43]: @yurishkuro Do you want me to keep the comment before the ones which the formatter doesn't cause an issue

Comment (yurishkuro) [reply cmd/collector/app/sanitizer/service_name_sanitizer_test.go:43]: no, it's fine. I am not ignoring this PR, it's just too large, I am reviewing it in several tries.

One thing that I found puzzling is why you sometimes use `_`, sometimes no-name, and sometimes replace `_` with no-name? My preference would be to have fewer variations, e.g.
  * always use `_` to indicate unused parameter (I find it more readable this way then just a type)
  * when primitive types are used, add a comment with parameter name

Comment (FlamingSaint) [reply cmd/collector/app/sanitizer/service_name_sanitizer_test.go:43]: > no, it's fine. I am not ignoring this PR, it's just too large, I am reviewing it in several tries.

Sorry, I was just curious if I needed to change something or did something wrong. 

> One thing that I found puzzling is why you sometimes use `_`, sometimes no-name, and sometimes replace `_` with no-name? My preference would be to have fewer variations, e.g.
> 

I use ( _ ) only if some of the parameters not used in the function. If all the parameters are not used I remove all the parameter names instead of using ( _ ).

Comment (FlamingSaint) [reply cmd/collector/app/sanitizer/service_name_sanitizer_test.go:43]: > * always use `_` to indicate unused parameter (I find it more readable this way then just a type)
> * when primitive types are used, add a comment with parameter name

a) This works as well, let me know if I need to make this change
b) I think this is already done

Comment (yurishkuro) [reply cmd/collector/app/sanitizer/service_name_sanitizer_test.go:43]: > let me know if I need to make this change

at this point I would just leave it as is. I don't want to re-review files I already reviewed.

Comment (FlamingSaint) [reply cmd/collector/app/sanitizer/service_name_sanitizer_test.go:43]: I guess I should have gone with the ( _ ) for everything from the start. My bad.

**Review by yurishkuro — APPROVED** [review]
